backend/analytics_server/mhq/exapi/models/github_timeline.py

```diff
@@ -0,0 +1,132 @@
+from datetime import datetime
+from dataclasses import dataclass
+from typing import Any, Optional, Dict, cast
+
+
+from mhq.exapi.schemas.timeline import (
+    GitHubPullTimelineEvent,
+)
+from mhq.store.models.code.enums import PullRequestEventType
+from mhq.utils.log import LOG
+from mhq.utils.time import dt_from_iso_time_string
+
+
+@dataclass
+class GithubPullRequestTimelineEvents:
+    EVENT_CONFIG = {
+        "reviewed": {
+            "actor_path": "user",
+            "timestamp_field": "submitted_at",
+            "id_path": "id",
+        },
+        "ready_for_review": {
+            "actor_path": "actor",
+            "timestamp_field": "created_at",
+            "id_path": "id",
+        },
+        "commented": {
+            "actor_path": "user",
+            "timestamp_field": "created_at",
+            "id_path": "id",
+        },
+        "committed": {
+            "actor_path": "author.name",
+            "timestamp_field": "author.date",
+            "id_path": "sha",
+        },
+        "default": {
+            "actor_path": "actor",
+            "timestamp_field": "created_at",
+            "id_path": "id",
+        },
+    }
```

Comment:
there are too many chances of error over here with string dict keys and everything being a string. can we make this more robust ? `actor_path`, `timestamp_field` and `id_path` are common in all